apps/explorer/src/hooks/useSearch.ts

```diff
@@ -0,0 +1,98 @@
+        ['search', query],
+        () => {
+            const promises = [];
+
+            if (isValidTransactionDigest(query)) {
+                promises.push(async () => {
+                    const txdata: SuiTransactionResponse =
+                        await rpc.getTransactionWithEffects(query);
+                    return [
+                        {
+                            label: 'transaction',
+                            results: [
+                                {
+                                    id: txdata.certificate.transactionDigest,
+                                    label: txdata.certificate.transactionDigest,
+                                    type: 'transaction',
+                                },
+                            ],
+                        },
+                    ];
+                });
+            }
+
+            if (isValidSuiAddress(query) && !isGenesisLibAddress(query)) {
+                promises.push(async () => {
+                    const [from, to] = await Promise.all([
+                        rpc.getTransactions({ FromAddress: query }, null, 1),
+                        rpc.getTransactions({ ToAddress: query }, null, 1),
+                    ]);
+
+                    if (from.data[0] || to.data[0]) {
+                        return [
+                            {
+                                label: 'address',
+                                results: [
+                                    {
+                                        id: query,
+                                        label: query,
+                                        type: 'address',
+                                    },
+                                ],
+                            },
+                        ];
+                    } else {
+                        throw new Error('not a valid address');
+                    }
+                });
+            }
+
+            const normalized = normalizeSuiObjectId(query);
+            if (isValidSuiObjectId(normalized)) {
+                promises.push(async () => {
+                    const { details, status } = await rpc.getObject(normalized);
+
+                    if (is(details, SuiObject) && status === 'Exists') {
+                        return [
+                            {
+                                label: 'object',
+                                results: [
+                                    {
+                                        id: details.reference.objectId,
+                                        label: details.reference.objectId,
+                                        type: 'object',
+                                    },
+                                ],
+                            },
+                        ];
+                    }
+                });
+            }
+            return Promise.any(promises.map((p) => p())).catch(() => []);
+        },
+        {
+            enabled: !!query,
```

Comment:
Maybe add cache time here so that we don't keep this in cache for a long time, given it's just for searching

Reply:
added cache time of 2s - totally arbitrary